### PR TITLE
Include default system code page in default encodings

### PR DESCRIPTION
**Pull request by qgppl**
## Proposed changes

Include default system code page in default encodings list.

### Before

On Windows 10 PL, the default encoding set in GitExtensions contain the following: UTF-8, US-ASCII, Unicode, Central European (DOS).

"Central European DOS", cp852, is the one returned by GetEncoding(CultureInfo.CurrentCulture.TextInfo.OEMCodePage).

Under Windows however, the default 8-bit encoding of files in PL locale is "Central European (Windows)" (ansi / windows-1250). This encoding is returned by GetEncoding(0). So, in order to use it (in diff viewer, for example), one must find and change the git encoding setting.

### After

Result of GetEncoding(0) is also added to populate the default encoding list.

## Test methodology 

- Manual test run without configuration file
- Code is under try/catch so possible errors will be silently ignored.
- If GetEncoding for OEMCodePage and 0 return the same encoding, they will be deduplicated by map.

## Test environment(s) 

- Windows 10 PL

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by vbjay** [comment]
Reading from system level config and not global  and lower would also prevent us from missing the system level config where this encoding is usually configured.

File: GitCommands/Settings/AppSettings.cs [hunk 1]
```diff
@@ -2108,6 +2108,15 @@ void AddEncodingByName(string s)
...
                     // there are CultureInfo values without a code page
                 }
+
+                try
+                {
+                    AddEncoding(Encoding.GetEncoding(0));
+                }
+                catch
+                {
+                    // catch if no encoding for default CP found
```
Comment (pmiossec) [inline GitCommands/Settings/AppSettings.cs:2118]: Please replace CP by "code page".

**Review by pmiossec — APPROVED** [review]

**Comment by gerhardol** [comment]
Probably fine.
It is possible to configure this too:
image

If set, this patch will not be used

File: contributors.txt [hunk 2]
```diff
@@ -202,4 +202,5 @@ YYYY/MM/DD, github id, Full name, email
 2023/07/16, e-stadnik, Evgeny Stadnik ev.stadnik(at)outlook.com
 2023/10/20, dmitrybozhenok, Dmitry Bozhenok, dmitry.bozhenok(at)gmail.com
 2023/11/30, snelltheta, Steve Samons, snelltheta(at)gmail.com
-2023/12/01, pmgiant, Pawel Marchewka, ppmmggiiaanntt(at)gmail.com
\ No newline at end of file
+2023/12/01, pmgiant, Pawel Marchewka, ppmmggiiaanntt(at)gmail.com
+2024/01/01, qgppl, Grzegorz Pachocki, g.pachocki@wp.pl
```
Comment (RussKie) [inline contributors.txt:206]:
```suggestion
2024/01/01, qgppl, Grzegorz Pachocki, g.pachocki(at)wp.pl
```

**Review by RussKie — APPROVED** [review]
👍 modulo the other comments

File: GitCommands/Settings/AppSettings.cs [hunk 3]
```diff
@@ -2108,6 +2108,15 @@ void AddEncodingByName(string s)
                 {
                     // there are CultureInfo values without a code page
                 }
+
+                try
+                {
+                    AddEncoding(Encoding.GetEncoding(0));
```
Comment (vbjay) [inline GitCommands/Settings/AppSettings.cs:2114]: Does AddEncodings prevent duplicate encodings?  If default is one that is also added by the other work then we will potentially get same encoding twice.

**Review by vbjay — COMMENTED** [review]
Potential duplicate encodings being added to list.

File: GitCommands/Settings/AppSettings.cs (hunk 3 above)
Comment (gerhardol) [reply GitCommands/Settings/AppSettings.cs:2114]: `Dictionary<string, Encoding> AvailableEncodings` so this should be OK

Comment (qgppl) [reply GitCommands/Settings/AppSettings.cs:2114]: Thanks, @gerhardol, exactly that.

**Review by mstv — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**